File: src/main/java/diceRollSimulator/pojos/DiceRollList.java
package diceRollSimulator.pojos;

import org.springframework.stereotype.Component;

import java.util.ArrayList;
import java.util.List;

@Component
public class DiceRollList {
    private final List<DiceRoll> diceRolls;

    public DiceRollList(){
        this.diceRolls = new ArrayList<>();
    }

    public List<DiceRoll> getDiceRolls(){
        return this.diceRolls;
    }

    public void addDiceRoll(DiceRoll diceRoll){
        this.diceRolls.add(diceRoll);
    }

    public void removeDiceRoll(int index) { this.diceRolls.remove(index); }

    public void clear() { this.diceRolls.clear(); }
}

File: src/main/frontend/src/components/DiceRollList.js
import React from 'react'
import DiceRoll from "../js/diceRollSimulator/DiceRoll";
import DiceRollInfo from "./DiceRollInfo";

function DiceRollList(props){

    return (props.diceRolls.length ?(
                <div>
                    {
                        props.diceRolls.map(diceRoll=> (
                            <div>
                                <DiceRollInfo diceRoll = {diceRoll} />
                                <hr/>
                            </div>
                            )
                        )
                    }
                </div>
            ):
            (
                <div>
                    <p> List empty </p>
                </div>
            )
    );
}

export default DiceRollList;